src/openscad.cc

```diff
@@ -320,46 +331,89 @@ int cmdline(const char *deps_output_file, const std::string &filename, const cha
 	}
 
 	FileModule *root_module;
-	ModuleInstantiation root_inst("group");
-	const AbstractNode *root_node;
-	AbstractNode *absolute_root_node;
-	shared_ptr<const Geometry> root_geom;
 
-	handle_dep(filename);
+	handle_dep(cmd.filename);
 
-	std::ifstream ifs(filename.c_str());
+	std::ifstream ifs(cmd.filename.c_str());
 	if (!ifs.is_open()) {
-		PRINTB("Can't open input file '%s'!\n", filename.c_str());
+		PRINTB("Can't open input file '%s'!\n", cmd.filename.c_str());
 		return 1;
 	}
 	std::string text((std::istreambuf_iterator<char>(ifs)), std::istreambuf_iterator<char>());
 	text += "\n\x03\n" + commandline_commands;
-	if (!parse(root_module, text, filename, filename, false)) {
+	if (!parse(root_module, text, cmd.filename, cmd.filename, false)) {
 		delete root_module;  // parse failed
 		root_module = nullptr;
 	}
 	if (!root_module) {
-		PRINTB("Can't parse file '%s'!\n", filename.c_str());
+		PRINTB("Can't parse file '%s'!\n", cmd.filename.c_str());
 		return 1;
 	}
 
 	// add parameter to AST
 	CommentParser::collectParameters(text.c_str(), root_module);
-	if (!parameterFile.empty() && !setName.empty()) {
+	if (!cmd.parameterFile.empty() && !cmd.setName.empty()) {
 		ParameterSet param;
-		param.readParameterSet(parameterFile);
-		param.applyParameterSet(root_module, setName);
+		param.readParameterSet(cmd.parameterFile);
+		param.applyParameterSet(root_module, cmd.setName);
 	}
-    
+
 	root_module->handleDependencies();
 
-	auto fpath = fs::absolute(fs::path(filename));
+	auto fpath = fs::absolute(fs::path(cmd.filename));
 	auto fparent = fpath.parent_path();
 	fs::current_path(fparent);
 	top_ctx->setDocumentPath(fparent.string());
 
 	AbstractNode::resetIndexCounter();
-	absolute_root_node = root_module->instantiate(top_ctx.ctx, &root_inst, nullptr);
+
+	if (cmd.animate_frames == 0) {
+		return do_export(cmd, tree, top_ctx, curFormat, root_module);
+	}
+	else {
+		// export the requested number of animated frames
+		for (unsigned frame = 0; frame < cmd.animate_frames; ++frame) {
```

Comment:
This never gets to $t = 1.0.  This works great for 360 degree animations but not so well if you want to rotate from -45 to 45 degrees.